include/dlaf/eigensolver/bt_band_to_tridiag/impl.h

```diff
@@ -259,13 +259,101 @@ struct TileAccessHelper {
   Part part_top_;
   Part part_bottom_;
 };
+
+template <class T, Device D>
+auto splitTileAndDiscard(pika::future<matrix::Tile<T, D>> tile,
+                         const matrix::SubTileSpec spec) noexcept {
+  return splitTile(tile, spec);
+}
+
```

Comment:
This helper resulted useful whenever I wanted to split a readwrite tile, but I don't care about the "joined" future after the splitted parts are consumed.

It is very specific and targets exactly the use case I need here.

Waiting for your comments before moving it to a more generic place, and possibly extending it to other API signatures (if you think it is worth).

Reply:
How does this work differently than just calling `splitTile`? Is the difference that `splitTile` takes a reference to a tile and therefore you can't call it directly? And finally, is it still appropriate to have `splitTile` in the name if the original is discarded?

Reply:
> Is the difference that `splitTile` takes a reference to a tile and therefore you can't call it directly?

You're right, it's exactly the point!



> And finally, is it still appropriate to have `splitTile` in the name if the original is discarded?

Probably it's not the best name, but IMHO *splitTile* still makes sense, because it refers to the action of spltting a tile in subtiles, not to what happens to the coming next future, for which I suffixed "discard" to describe this last behavior.

Just to point out that, it actually splits the tile, but it does not give you back the access to the immediately next tile ("discard" refers to this behavior), because its ownership is not kept by "anyone". However, it does not mean we won't be able to access anymore that tile, because we can always access the original source of the tile, i.e. the chain correctness is ensured.


I'm going to open a small PR with this change isolated, because it may need a bit of discussion. I will link that here too, as soon as I open it 😉

Reply:
Here it is #504

Reply:
It has been rebased and adapted since #504 got merged.